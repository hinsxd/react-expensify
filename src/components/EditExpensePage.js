import React, { Component } from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends Component {
  onSubmit = expense => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Edit Expense</h1>
        <p>Editing {this.props.expense.id}</p>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToObject = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});
const mapDispatchToProps = (dispatch, props) => ({
  editExpense: expense => dispatch(editExpense(expense.id, expense)),
  removeExpense: data => dispatch(removeExpense(data))
});

export default connect(
  mapStateToObject,
  mapDispatchToProps
)(EditExpensePage);
