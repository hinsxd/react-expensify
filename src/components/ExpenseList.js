import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
export const ExpenseList = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>No Expense </p>
    ) : (
      props.expenses.map(expense => (
        <ExpenseListItem key={expense.id} {...expense} />
      ))
    )}
    {}
  </div>
);

const mapStateToProps = state => ({
  expenses: selectExpenses(state)
});

export default connect(mapStateToProps)(ExpenseList);
