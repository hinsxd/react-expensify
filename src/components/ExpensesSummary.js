import React from 'react';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import { connect } from 'react-redux';
export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
      </h1>
    </div>
  );
};
const mapStateToProps = state => ({
  expensesTotal: selectExpensesTotal(selectExpenses(state)),
  expensesCount: selectExpenses(state).length
});

export default connect(mapStateToProps)(ExpensesSummary);
