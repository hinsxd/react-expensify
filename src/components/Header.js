import React from 'react';
import { NavLink } from 'react-router-dom';
const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-acrtive" exact={true}>
      Home
    </NavLink>
    <NavLink to="/create" activeClassName="is-acrtive">
      Create Expense
    </NavLink>
  </header>
);

export default Header;
