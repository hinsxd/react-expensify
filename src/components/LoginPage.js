import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin, startLogout } from '../actions/auth';

export const LoginPage = ({ startLogin, startLogout }) => (
  <div>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin()),
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
