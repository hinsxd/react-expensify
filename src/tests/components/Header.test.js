import React from 'react';
import { Header } from '../../components/Header';
import { shallow } from 'enzyme';
// react-test-renderer

test('should render header correctly', () => {
  const wrapper = shallow(<Header startLogOut={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
