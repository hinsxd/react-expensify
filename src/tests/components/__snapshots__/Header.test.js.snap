// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-acrtive"
    aria-current="page"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-acrtive"
    aria-current="page"
    to="/create"
  >
    Create Expense
  </NavLink>
  <button>
    Logout
  </button>
</header>
`;
